src/conn/conn_compact.c

```diff
@@ -728,9 +728,10 @@ __wti_background_compact_server_create(WT_SESSION_IMPL *session)
 
     /*
      * Compaction does enough I/O it may be called upon to perform slow operations for the block
-     * manager.
+     * manager. Don't let the background compaction thread be pulled into eviction to limit
+     * performance impact, when the cache is under pressure, compaction should halt straight away.
```

Comment:
I don't think we should get into the implementation details here:

```suggestion
     * performance impacts.
```